client.go

```diff
@@ -306,11 +306,65 @@ func (c *client) unassignRoles(ctx context.Context, roleIDs []string) error {
 	return merr.ErrorOrNil()
 }
 
+// addGroupMemberships adds the service principal to the Azure groups.
+func (c *client) addGroupMemberships(ctx context.Context, sp *graphrbac.ServicePrincipal, groups []*AzureGroup) ([]string, error) {
+	groupsIDs := make([]string, 0, len(groups))
```

Comment:
Is returning this just a convenience, or is there some cases in which the information is different than what the caller already has?

Reply:
Yeah it was just for convenience. Pulled it out into a dedicated helper